src/PolygonDraw/usePolygonEditor.ts

```diff
@@ -0,0 +1,100 @@
+import { useReducer, useEffect, useMemo, useCallback, useRef } from 'react';
+import { Coordinate } from 'types';
+import isEqual from 'lodash.isequal';
+
+import { actions } from './actions';
+import { ensurePolygonList, isPolygonClosed } from '../helpers';
+import { polygonEditReducer, initialState } from './reducer';
+import { isValidPolygon } from './validators';
+
+function usePrevious(value: Coordinate[][]) {
+    const ref = useRef<Coordinate[][]>();
+    useEffect(() => {
+        ref.current = value;
+    }, [value]);
+    return ref.current;
+}
+
+export const usePolygonEditor = (
+    onChange: (polygon: Coordinate[], isValid: boolean) => void = () => {},
+    polygons: Coordinate[] | Coordinate[][],
+    activeIndex: number
+) => {
+    const polygonList = ensurePolygonList(polygons);
+
+    const [state, dispatch] = useReducer(polygonEditReducer, initialState, init => {
+        return {
+            ...init,
+            activeIndex: activeIndex,
+            polygons: polygonList
+        };
+    });
+
+    const activePolygon = useMemo(() => state.polygons[state.activeIndex], [state.polygons, state.activeIndex]);
+    const polygonIsClosed: boolean = useMemo(() => isPolygonClosed(activePolygon), [activePolygon]);
+
+    const prevPolygons = usePrevious(polygonList);
+
+    useEffect(() => {
+        if (!isEqual(prevPolygons, polygonList)) {
+            dispatch(actions.changePolygon(polygonList));
+        }
+    }, [polygonList]);
+
+    useEffect(() => {
+        if (onChange) {
```

Comment:
**Change**:
I think we won't need this since the onChange argument has a fallback